documentation/docs/developer_guide/coding_standards.md

```diff
@@ -1 +1,179 @@
+
+Currently not all of the CABLE code is compliant with the standards described here. Work is in progress to apply these standards across the existing code base, however these standards are expected to be met by any new development.
+
+Whilst syntax/text formatting style is relatively trivial, it must still be adhered to. Some more significant general rules follow.
+
+## General Rules
+
+### 1. IO must be restricted to the top level
+
+In terms of the offline application of the model this means IO should be restricted to the `offline/` directory. There should be no IO code elsewhere in the CABLE directory. 
+
+### 2. Comment liberally
+
+Detrimental to any software is the lack of meaningful and/or outdated comments. 
+Liberal inline commenting is expected, as well as use of Markdown for FORD generated documentation. See [cable_roughness.F90](https://github.com/CABLE-LSM/CABLE/blob/main/src/science/roughness/cable_roughness.F90) for an example, which is then redered [here](https://cable.readthedocs.io/en/latest/api/module/cable_roughness_module.html)
+
+### 3. Avoid WHERE loops
+
+`WHERE` loops are sometimes used in stead of `DO` loops with an `IF`. In some circumstances, this may improve readability and is easier to write. However, they are difficult to debug, hamper portability and are therefore discouraged. 
+Moreover it will not pass JULES coding standards. 
+
+### 4. Variable names
+
+We are not so limited by the number of character spaces available as we once were. Developers are encouraged to name variables meaningfully. Commenting might make it clear what the variable corresponds to, however, it is helpful to be able to understand what is going on whilst reading a line of code.
+
+Please maintain globally consistent names of variables across files, subroutines and modules. 
+
+### 5. File names
+
+To include CABLE code identifiably across applications, please name the file with a `_cbl` suffix in your filename.
+For example: `egName_cbl.F90`. 
+
+Please maintain consistent naming between files, subroutines and modules. 
+
+### 6. MODULE names
+
+Everything should be contained within modules.
+Modules should be named of the form `*_mod_cbl`. Following from the above example, the file `egName_cbl.F90` should contain the module declaration `MODULE egName_mod_cbl`
+
+### 7. SUBROUTINE names
+
+Subroutines should be of the form "\*_cbl()". Following from the above file `egName_cbl.F90` `SUBROUTINE egName_cbl()`. This is not only helpful for consistency but necessary within JULES/UM applications where it distinguishes the CABLE version of an analogous JULES subroutine. 
+Markdown needs to escape * characters.
```

Comment:
We are not using this convention in the code we submitted to JaC. So I would remove unless we want to rename everything.
```suggestion
### 7. SUBROUTINE names

Subroutines should be of the form "\*_cbl()". Following from the above file `egName_cbl.F90` `SUBROUTINE egName_cbl()`. This is not only helpful for consistency but necessary within JULES/UM applications where it distinguishes the CABLE version of an analogous JULES subroutine. 
```